### PR TITLE
feat: governance contract

### PR DESCRIPTION
**Pull request by eomti-wm**
**draft version**
- implement governance contract
  - `GovStaking`
  - `GovNCP`

-  ~~fix `blockPeriod` bug in the `AdjustTime()` of simulated-backend.~~

### CODE REVIEW

File: wemixgov/governance-contract/contracts-wbft/GovNCP.sol [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        _proposal.decisions[msg.sender] = _decision;
+
+        emit Vote(_proposalID, msg.sender, _accept);
+        uint256 _threshold = __ncpList.length();
+        if (_proposal.accepts * 2 > _threshold || _proposal.rejects * 2 >= _threshold) {
+            _finalizeProposal(_proposal, _proposal.accepts > _proposal.rejects);
+        }
+    }
+
+    function cancelProposal(uint256 _proposalID) external onlyNCP {
```
Comment (egonspace) [inline wemixgov/governance-contract/contracts-wbft/GovNCP.sol:104]: Shouldn't it only be possible to cancel the currently ongoing proposal? If so, we must not receive the proposal id.

Comment (egonspace) [reply wemixgov/governance-contract/contracts-wbft/GovNCP.sol:104]: I found that `_getVotingProposal` defends that case. It's resolved

File: wemixgov/governance-contract/contracts-wbft/GovStaking.sol [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+import "@openzeppelin/contracts/utils/Address.sol";
+import "./GovConst.sol";
+
+contract GovStaking {
+    using EnumerableSet for EnumerableSet.AddressSet;
+    using Address for address payable;
+
+    struct Validator {
+        address staker;
+        address reward;
```
Comment (jed-wemade) [inline wemixgov/governance-contract/contracts-wbft/GovStaking.sol]: To reduce ambiguity, `rewardee` is better than `reward`.

**Review by egonspace — APPROVED** [review]

File: wemixgov/governance-wbft/contracts.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package govwbft
+
+import _ "embed"
+
+// contract codes for Chapel upgrade
```
Comment (jed-wemade) [inline wemixgov/governance-wbft/contracts.go:5]: Chapel is testnet name for BNB Chain.

File: wemixgov/governance-wbft/staking.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	SLOT_TOTAL_STAKING    = "0x0"
+	SLOT_STAKER_SET       = "0x1" // ,0x2
+	SLOT_STAKER_INFO      = "0x3"
+	SLOT_STAKER_BY_STAKER = "0x4"
```
Comment (jed-wemade) [inline wemixgov/governance-wbft/staking.go:13]: `SLOT_STAKER_BY_OPERATOR`

**Pull request merged**